### DISCUSSION TITLE
How to hardcode a custom DNS into New peer(client)

### DISCUSSION THREAD

**Discussion by sigma2017** [wg-easy/wg-easy#765] [General]
Hello,
I am using wireguard for a while and was looking for a easy solution and found this wg-easy with docker.
I can say that I am impressed, it is working like a charm!
However, on the same VPS where wg-easy is running, I also have adguard listening on 5353 docker based with filters, custom rules and so on and I need wg to resolv any dns using that adguard container.

I make it work starting wg-easy with --dns 172.17.0.1 and once started, if for example I try   docker exec -it wg-easy nslookup yahoo.fr I got Server:         172.17.0.1, Address:        172.17.0.1:53 and indeed  adguard is by default used to resolve any domain.

I have created a peer, put it via QR code on my samsung phone but the conf file has either 1.1.1.1 or 8.8.8.8 as DNS. I need to edit it and replace it with 172.17.0.1 and then to connect.

Is there any method to automatically create the client conf with DNS 172.17.0.1 instead of 1.1.1.1 or 8.8.8.8 ??

Thanks in advance!
-----------------------
Later Edit:
  -e WG_DEFAULT_DNS=172.17.0.1 \
this solved the issue, new client conf is automatically created with the custom DNS.
Amazing solution,
Thank you!

**Comment by vit5041** [comment]
v15 does not use env vars to configure dns. you have to do it in the admin panel

https://github.com/wg-easy/wg-easy/issues/2020#issuecomment-3067908271
